### ISSUE TITLE
[clang-tidy] Fails to detect readability-identifier-naming.TypedefPrefix check in case of macro usage

### ISSUE THREAD

**Issue by TyborAUT** [llvm/llvm-project#55156]
Tested with 13.0.0 and 14.0.0 and does not detect invalid case style for typedef!

Command line: `clang-tidy --config="{Checks: '-*,readability-identifier-naming', CheckOptions: [{key: readability-identifier-naming.TypedefPrefix, value: t_}]}" clang-tidy_typedef_bug.c`

When macro `STATIC_MACRO` is defined as `static` no warning is issued. When macro `STATIC_MACRO` is empty the check finds the invalid case style!

```c
typedef enum {
    e_value_0,
    e_value_1,
    e_value_2,
    e_value_3,
    e_value_4,
    e_value_5
} e_values_type;

#define STATIC_MACRO static

STATIC_MACRO void some_func(e_values_type values)
{
    // do something
}
```
